libraries/AP_GPS/GPS_Backend.cpp

```diff
@@ -353,8 +353,20 @@ bool AP_GPS_Backend::calculate_moving_base_yaw(const float reported_heading_deg,
 
 #ifndef HAL_BUILD_AP_PERIPH
         {
-            const Vector3f antenna_tilt = AP::ahrs().get_rotation_body_to_ned() * offset;
+            // get lag
+            float lag = 0.1;
+            get_lag(lag);
+
+            // get vehicle rotation, projected back in time using the gyro
+            auto &ahrs = AP::ahrs();
+            const Vector3f &gyro = ahrs.get_gyro();
+            Matrix3f rot_body_to_ned = ahrs.get_rotation_body_to_ned();
+            rot_body_to_ned.rotate(gyro * (-lag));
```

Comment:
This assumes that we have not changed our yaw rate in the last lagged period of time doesn't it? I agree it's generally going to be correct, but it seems like it would go wrong on a very agile platform, such as a copter. Could the EKF be managing this better internally, with the IMU sample buffers?

Reply:
> Could the EKF be managing this better internally, with the IMU sample buffers?

it could, but it would be a much more complex change. We'd need an AHRS/EKF API to ask for the attitude at a past time. The EKF does know that information (although DCM doesn't), so it is doable, but adds complexity.
This fix involves no API changes, and I think will do well in almost all cases, especially for the larger vehicles which tend to use moving baseline GPS.